coderd/database/migrations/000274_workspace_modules.up.sql

```diff
@@ -0,0 +1,14 @@
+ALTER TABLE
+    workspace_resources
+ADD
+    COLUMN module_path TEXT;
+
+CREATE TABLE workspace_modules (
+    id uuid NOT NULL,
+    job_id uuid NOT NULL REFERENCES provisioner_jobs (id) ON DELETE CASCADE,
+    transition workspace_transition NOT NULL,
+    source TEXT NOT NULL,
+    version TEXT NOT NULL,
+    key TEXT NOT NULL,
+    created_at timestamp with time zone NOT NULL
+);
```

Comment:
Why do we need to insert these to the database? Is there any other reason we might want to save this data, or is it only for telemetry?

Reply:
This is only for telemetry purposes. Since `CompleteJob` reports telemetry after finishing, we could skip saving the data in the database. The only downside is that if the `CompleteJob` report fails for any reason, we won't be able to retry sending it later.

I don't have a strong opinion on this - if you prefer we skipped saving, I'll be happy to do that.

Reply:
This also had me scratching my head.

> The only downside is that if the CompleteJob report fails for any reason, we won't be able to retry sending it later.

_Do_ we retry if telemetry fails to send?

Reply:
This is my first time really delving into the telemetry stuff, but here's what I think is true at the time of writing (please correct me if I'm wrong):

1) Telemetry snapshots are done periodically so any data we do not persist will be lost between coderd restarts
2) Telemetry snapshots are done by each coderd independently, so data that is known to one `coderd` only may not be present in snapshots from other coderd instances.

I'm not sure if there is any sort of de-duplication logic on the telemetry server end, but having to merge multiple disparate snapshots sounds like it would get messy and error-prone.

All of that is basically to say: we probably do need to store in the database in order to get a consistent snapshot.

Reply:
@dannykopping we run a periodic snapshot of a subset of tables in the db created [within the last hour](https://github.com/coder/coder/blob/ea2b6b7cbc9022ec92ee49f74c75aefd77613360/coderd/telemetry/telemetry.go#L292). So we actually store multiple instances of rows with the same id in our telemetry db and deduplicate when querying the data. That's what I meant by retrying - if the `CompleteJob` snapshot fails to send, the next periodic snapshot will send the same data again.

@johnstcn as [discussed on Slack](https://codercom.slack.com/archives/CJURPL8DN/p1731322612970969) storing the data in PG may not be required for consistency, but storing the module data will likely come in handy elsewhere, so it's worth doing anyway.

Reply:
@dannykopping Do you mean storing the provider info in the database, telemetry, or both? This is likely out of scope for this PR - would you like to create an issue for it?

Reply:
Sorry for muddying the waters, let's not worry about providers for now.